libsplinter/src/biome/credentials/rest_api/actix_web_1/mod.rs

```diff
@@ -0,0 +1,266 @@
+    pub fn get_identity_provider(&self) -> BiomeUserIdentityProvider {
+        BiomeUserIdentityProvider::new(
+            self.token_secret_manager.clone(),
+            default_validation(&self.credentials_config.issuer()),
+        )
+    }
+}
+
+impl RestResourceProvider for BiomeCredentialsRestResourceProvider {
+    fn resources(&self) -> Vec<Resource> {
+        let mut resources = Vec::new();
+
+        resources.push(user::make_list_route(self.credentials_store.clone()));
+        resources.push(verify::make_verify_route(
+            self.credentials_store.clone(),
+            self.credentials_config.clone(),
+            self.token_secret_manager.clone(),
+        ));
+        resources.push(login::make_login_route(
+            self.credentials_store.clone(),
+            self.refresh_token_store.clone(),
+            self.credentials_config.clone(),
+            Arc::new(AccessTokenIssuer::new(
+                self.token_secret_manager.clone(),
+                self.refresh_token_secret_manager.clone(),
+            )),
+        ));
+        resources.push(token::make_token_route(
+            self.refresh_token_store.clone(),
+            self.token_secret_manager.clone(),
+            self.refresh_token_secret_manager.clone(),
+            Arc::new(AccessTokenIssuer::new(
+                self.token_secret_manager.clone(),
+                self.refresh_token_secret_manager.clone(),
+            )),
+            self.credentials_config.clone(),
+        ));
+        resources.push(logout::make_logout_route(
+            self.refresh_token_store.clone(),
+            self.token_secret_manager.clone(),
+            self.credentials_config.clone(),
+        ));
+
+        resources.push(register::make_register_route(
+            self.credentials_store.clone(),
+            self.credentials_config.clone(),
+        ));
+
+        #[cfg(feature = "biome-key-management")]
+        {
+            resources.push(user::make_user_routes(
+                self.credentials_config.clone(),
+                self.credentials_store.clone(),
+                self.key_store.clone(),
+            ));
+        }
+
+        resources
+    }
+}
+
+/// Builder for BiomeCredentialsRestResourceProvider
+#[derive(Default)]
+pub struct BiomeCredentialsRestResourceProviderBuilder {
+    #[cfg(feature = "biome-key-management")]
+    key_store: Option<Arc<dyn KeyStore>>,
+    credentials_config: Option<BiomeCredentialsRestConfig>,
+    token_secret_manager: Option<Arc<dyn SecretManager>>,
+    refresh_token_secret_manager: Option<Arc<dyn SecretManager>>,
+    refresh_token_store: Option<Arc<dyn RefreshTokenStore>>,
+    credentials_store: Option<Arc<dyn CredentialsStore>>,
+}
+
+impl BiomeCredentialsRestResourceProviderBuilder {
+    /// Sets a KeyStore for the BiomeCredentialsRestResourceProvider
+    ///
+    /// # Arguments
+    ///
+    /// * `pool`: ConnectionPool to database that will serve as backend for KeyStore
```

Comment:
The arg in this comment doesn't match the function signature

Reply:
Fixed